### DISCUSSION TITLE
How to implement TreeView with dynamic loading and search together?

### DISCUSSION THREAD

**Discussion by hbthanki** [NG-ZORRO/ng-zorro-antd#7346] [Q&A]
How to implement TreeView with dynamic loading and search together?
From the documentation(examples), I can see both Dynamic(fetching) and Search are using different datasources. Is it possible to combine both functionalities in one component?
I have a requirement where I need to give search and dynamic loading of children.
@hsuanxyz, @vthinkxie Please can you help!
TIA.

**Comment by hsuanxyz** [comment]
This is technically possible, but the exact implementation depends on

- whether to search for both leaf nodes and non-leaf nodes
- whether to allow expansion in the search results
- The design of the backend API (e.g. whether the input requires a parent id, whether the returned result contains hierarchical information).

The easiest way is to use two different datasources, but the UI looks like one.

**Reply by hbthanki** [reply]
Thank you so much for your prompt reply. Do you have any examples where two different datasources are used?
I have a requirement where user right clicks and it fetches children nodes for the selected node. And (s)he can search through the tree.

**Reply by hbthanki** [reply]
@hsuanxyz 
Does this mean that for the same component, two features can not be used together?

**Reply by MacherelR** [reply]
Hi @hbthanki, did you achieve that kind of implementation ?

**Reply by ysk3a** [reply]
i am curious about this as well. the treeview section of this library peaked my interest and this maybe some thing id like to add if it is possible.
